### PR TITLE
[GNA] Fixed handling of transposes around MatMul

### PR DESCRIPTION
**Pull request by elilobanova**
### Details:
 - Transposed after MatMul should be removed if there is an activation between them
 - Swap MatMul inputs transformation should handle a case when there are Transposes around MatMul, these transposes should be removed instead of adding new ones
 - Some more small fixes for Transposes handling
 - Tests are modified and added according to made changes

### Tickets:
56549

### CODE REVIEW

File: inference-engine/src/gna_plugin/backend/gna_limitations.hpp [hunk 1]
```diff
@@ -31,10 +31,16 @@ constexpr uint32_t maxPoolMaxWindowSize = 6;
 constexpr uint32_t copyMaxGrouping = 8;
 constexpr uint32_t transposeMaxSize = 65528;
 
+inline bool IsTranspose2d(const std::vector<size_t>& shape) {
+    auto shape_no_1 = shape;
+    shape_no_1.erase(std::remove(shape_no_1.begin(), shape_no_1.end(), 1), shape_no_1.end());
```
Comment (mryzhov) [inline inference-engine/src/gna_plugin/backend/gna_limitations.hpp:36]: maybe count or count_if instead of copying and erasing vector twice?

**Review by mryzhov — CHANGES_REQUESTED** [review]

**Review by mryzhov — COMMENTED** [review]
Could you please also update transformation description in header

File: inference-engine/src/gna_plugin/backend/gna_limitations.hpp (hunk 1 above)
Comment (elilobanova) [reply inference-engine/src/gna_plugin/backend/gna_limitations.hpp]: done

File: inference-engine/src/gna_plugin/transformations/swap_input_matmul_gna.cpp [hunk 2]
```diff
@@ -223,14 +267,115 @@ SwapInputMatMulWithFq::SwapInputMatMulWithFq() {
...
+        ngraph::pattern::wrap_type<ngraph::opset8::Constant>(),
+        ngraph::pattern::wrap_type<ngraph::opset8::Constant>(),
+        ngraph::pattern::wrap_type<ngraph::opset8::Constant>()});
+    auto act_input = std::make_shared<ngraph::pattern::op::Or>(ngraph::OutputVector{fq_input, fq});
+    auto act = ngraph::pattern::wrap_type<ngraph::opset8::Relu, ngraph::opset8::Sigmoid,
+        ngraph::opset8::Tanh, ngraph::opset8::Abs, ngraph::opset8::Log, ngraph::opset8::Exp,
+        ngraph::opset8::Sign, ngraph::opset8::Clamp>({act_input});
+    auto callback = [=](ngraph::pattern::Matcher& m) {
+        const auto& pattern_map = m.get_pattern_value_map();
+        bool first_input_const = true;
```
[the remark this answers is not in the document]
Comment (elilobanova) [reply inference-engine/src/gna_plugin/transformations/swap_input_matmul_gna.cpp]: done

File: inference-engine/src/gna_plugin/transformations/handle_transposes_around_matmul.cpp [hunk 3]
```diff
@@ -48,18 +48,35 @@ void InsertTranspose(std::shared_ptr<ngraph::Node> prev_node, const std::string&
...
+
     auto transpose_order = ngraph::opset8::Constant::create(ngraph::element::i64, ngraph::Shape{permute_order.size()}, permute_order);
-    auto transpose = std::make_shared<ngraph::opset8::Transpose>(prev_node, transpose_order);
-    transpose->set_friendly_name(base_name + "/in_transpose");
+    node = std::make_shared<ngraph::opset8::Transpose>(node, transpose_order);
+    node->set_friendly_name(base_name + "/in_transpose");
+    new_ops.push_back(node);
+
+    if (before_matmul) {
+        auto reshape_const_after = std::make_shared<ngraph::opset8::Constant>(ngraph::element::Type_t::i64,
```
[the remark this answers is not in the document]
Comment (elilobanova) [reply inference-engine/src/gna_plugin/transformations/handle_transposes_around_matmul.cpp]: done

**Review by nkogteva — APPROVED** [review]

**Review by nkogteva — APPROVED** [review]

**Review by mryzhov — APPROVED** [review]

**Pull request merged**